src/common/support_utils.cpp

```diff
@@ -164,3 +165,20 @@ void create_path_info_4service(char *str, const uint32_t str_size) {
     append_crc(str, str_size);
 #endif //0
 }
+
+bool appendix_exist() {
+    const version_t *bootloader = (const version_t *)BOOTLOADER_VERSION_ADDRESS;
+    GPIO_InitTypeDef GPIO_InitStruct = { 0 };
+    GPIO_InitStruct.Pin = GPIO_PIN_13;
+    GPIO_InitStruct.Mode = GPIO_MODE_INPUT;
+    GPIO_InitStruct.Pull = GPIO_NOPULL;
+
+    if (bootloader->major >= 1 && bootloader->minor >= 1) {
+        return !(ram_data_exchange.model_specific_flags & APPENDIX_FLAG_MASK);
+    } else {
+        HAL_GPIO_Init(GPIOA, &GPIO_InitStruct);
```

Comment:
something should revert changes made by `HAL_GPIO_Init(GPIOA, &GPIO_InitStruct);` after read

Reply:
also missing `#ifndef _DEBUG`

Reply:
HAL_GPIO_Init doesn't force any state of the PIN by GPIO_NOPULL. There is only read of the PIN setup by hardware solution of the appendix.